### PR TITLE
Cog 170 pgvector adapter

### PR DESCRIPTION
**Pull request by dexters1**
Add PGVector support for cognee

## Release Notes

- **New Features**
  - Added support for the "pgvector" vector database provider.
  - Introduced a new GitHub Actions workflow for automated testing of the pgvector integration.
  - Enhanced documentation in the README to include new vector storage options.
  - Expanded configuration options for vector databases in the settings.

- **Bug Fixes**
  - Improved clarity in configuration settings for various database providers.

- **Documentation**
  - Updated README with expanded installation instructions and additional supported technologies.
  
- **Chores**
  - Updated `docker-compose.yml` to use the new `pgvector` PostgreSQL image.
  - Modified `pyproject.toml` to include new dependencies for pgvector.

### CODE REVIEW

File: cognee/tests/test_pgvector.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import logging
+import pathlib
+import cognee
+from cognee.api.v1.search import SearchType
+
+logging.basicConfig(level=logging.DEBUG)
+
+async def main():
+    #TODO: Should this be set on pipeline side or in the test?
+    cognee.config.set_vector_engine_provider("pgvector")
```
Comment (borisarzentar) [inline cognee/tests/test_pgvector.py:13]: Let's refactor to have 1 method. Like `cognee.config.set_vector_db_config({...})` and `set_relational_db_config({...})`

File: cognee/infrastructure/databases/vector/pgvector/PGVectorAdapter.py [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+from pgvector.sqlalchemy import Vector
+
+from ...relational.sqlalchemy.SqlAlchemyAdapter import SQLAlchemyAdapter
+from ...relational.ModelBase import Base
+
+from datetime import datetime
+
+
+# TODO: Find better location for function
+def serialize_datetime(data):
```
Comment (borisarzentar) [inline cognee/infrastructure/databases/vector/pgvector/PGVectorAdapter.py:20]: Let's move this function outside in a separate file.

File: cognee/infrastructure/databases/vector/create_db_and_tables.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+from ..relational.ModelBase import Base
+from .get_vector_engine import get_vector_engine, get_vectordb_config
+from sqlalchemy import text
+
+async def create_db_and_tables():
```
Comment (borisarzentar) [inline cognee/infrastructure/databases/vector/create_db_and_tables.py:5]: We could try to move this inside pgvector adapter.

File: cognee/infrastructure/databases/vector/pgvector/PGVectorAdapter.py (hunk 2 above)
Comment (dexters1) [reply cognee/infrastructure/databases/vector/pgvector/PGVectorAdapter.py]: Done

File: cognee/infrastructure/databases/vector/create_db_and_tables.py (hunk 3 above)
Comment (dexters1) [reply cognee/infrastructure/databases/vector/create_db_and_tables.py]: Done

File: cognee/tests/test_pgvector.py (hunk 1 above)
Comment (dexters1) [reply cognee/tests/test_pgvector.py]: Done

File: cognee/infrastructure/databases/vector/pgvector/create_db_and_tables.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+from ...relational.ModelBase import Base
+from ..get_vector_engine import get_vector_engine, get_vectordb_config
+from sqlalchemy import text
+
+async def create_db_and_tables():
```
Comment (borisarzentar) [inline cognee/infrastructure/databases/vector/pgvector/create_db_and_tables.py:5]: If you merged with the `main`, you can see that `create_database` exists in `SqlAlchemyAdapter`. Is that something we can reuse here?

Comment (dexters1) [reply cognee/infrastructure/databases/vector/pgvector/create_db_and_tables.py:5]: Yes and no. By that I mean we can use create_database from SqlAlchemyAdapter as the initial step, but there is an extra step to enable pgvector for a postgres database that we have to run in an engine connection context

**Review by borisarzentar — APPROVED** [review]

**Pull request merged**